Kitodo/src/main/java/org/kitodo/production/services/data/ProcessService.java

```diff
@@ -920,6 +921,27 @@ private void convertRelatedJSONObjects(Map<String, Object> jsonObject, ProcessDT
         processDTO.setProgressInProcessing(getProgressInProcessing(null, processDTO.getTasks()));
         processDTO.setProgressOpen(getProgressOpen(null, processDTO.getTasks()));
         processDTO.setProgressLocked(getProgressLocked(null, processDTO.getTasks()));
+
+        if (processDTO.hasChildren()) {
+            List<Process> children;
+            try {
+                children = ServiceManager.getProcessService().getById(processDTO.getId()).getChildren();
+                processDTO.setProgressClosed(progressOfChildren(children, tasks -> getProgressClosed(tasks, null)));
+                processDTO.setProgressInProcessing(progressOfChildren(children, tasks -> getProgressInProcessing(tasks, null)));
+                processDTO.setProgressOpen(progressOfChildren(children, tasks -> getProgressOpen(tasks, null)));
+                processDTO.setProgressLocked(progressOfChildren(children, tasks -> getProgressLocked(tasks, null)));
+            } catch (DAOException dao) {
+                throw new DataException(dao);
+            }
+        }
+    }
+
+    private static Double progressOfChildren(List<Process> children, Function<List<Task>, Double> calculator) {
```

Comment:
Why are you handing the function here?
Please use:
`private Double progressOfChildren(List<Process> children) {
        DescriptiveStatistics statistics = new DescriptiveStatistics();
        for (Process child : children) {
            statistics.addValue(getProgressLocked(child.getTasks(), null));
        }
        return statistics.getMean();
    }`

Reply:
I wanted to deduplicate the code. What you want gives 4× the same function, just differing in the function called:

```java
            processDTO.setProgressClosed(progressOfChildrenClosed(children));
            processDTO.setProgressInProcessing(progressOfChildrenInProcessing(children));
            processDTO.setProgressOpen(progressOfChildrenOpen(children));
            processDTO.setProgressLocked(progressOfChildrenLocked(children));
        } catch (DAOException dao) {
            throw new DataException(dao);
        }
    }
}

private Double progressOfChildrenClosed(List<Process> children) {
    DescriptiveStatistics statistics = new DescriptiveStatistics();
    for (Process child : children) {
        statistics.addValue(getProgressClosed(child.getTasks(), null));
    }
    return statistics.getMean();
}

private Double progressOfChildrenInProcessing(List<Process> children) {
    DescriptiveStatistics statistics = new DescriptiveStatistics();
    for (Process child : children) {
        statistics.addValue(getProgressInProcessing(child.getTasks(), null));
    }
    return statistics.getMean();
}

private Double progressOfChildrenOpen(List<Process> children) {
    DescriptiveStatistics statistics = new DescriptiveStatistics();
    for (Process child : children) {
        statistics.addValue(getProgressOpen(child.getTasks(), null));
    }
    return statistics.getMean();
}

private Double progressOfChildrenLocked(List<Process> children) {
    DescriptiveStatistics statistics = new DescriptiveStatistics();
    for (Process child : children) {
        statistics.addValue(getProgressLocked(child.getTasks(), null));
    }
    return statistics.getMean();
}
```

Do you really want that?